### ISSUE TITLE
[libepoxy] fails to build -- blocked by upstream meson

### ISSUE THREAD

**Issue by Drako** [Microsoft/vcpkg#829]
The build fails at the stage

> CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:14 (message)
>     Command failed: C:/Users/drako/Desktop/Projekte/vcpkg/downloads/tools/python/python.exe;C:/Users/drako/Desktop/Projekte/vcpkg/downloads/tools/meson/meson-0.38.1/meson.py;--buildtype;plain;--backend;ninja;--default-library;shared;--prefix;C:/Users/drako/Desktop/Projekte/vcpkg/packages/libepoxy_x86-windows;C:/Users/drako/Desktop/Projekte/vcpkg/buildtrees/libepoxy/src/libepoxy-7d58fd3d47d2d69f2b1b9f08325302e4eeff9ebe
>     Working Directory: C:/Users/drako/Desktop/Projekte/vcpkg/buildtrees/libepoxy/x86-windows-rel
>     See logs for more information:
>       C:\Users\drako\Desktop\Projekte\vcpkg\buildtrees\libepoxy\config-x86-windows-rel-out.log
>       C:\Users\drako\Desktop\Projekte\vcpkg\buildtrees\libepoxy\config-x86-windows-rel-err.log

Content of the *-err.log is

> Exception in thread Thread-1:
> Traceback (most recent call last):
>   File "threading.py", line 914, in _bootstrap_inner
>   File "threading.py", line 862, in run
>   File "subprocess.py", line 1014, in _readerthread
>   File "encodings\cp1252.py", line 23, in decode
> UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 474: character maps to <undefined>
> 
> Exception in thread Thread-2:
> Traceback (most recent call last):
>   File "threading.py", line 914, in _bootstrap_inner
>   File "threading.py", line 862, in run
>   File "subprocess.py", line 1014, in _readerthread
>   File "encodings\cp1252.py", line 23, in decode
> UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 62: character maps to <undefined>
> 
> Traceback (most recent call last):
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\mesonmain.py", line 286, in run
>     app.generate()
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\mesonmain.py", line 160, in generate
>     intr = interpreter.Interpreter(b, g)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreter.py", line 1198, in __init__
>     self.parse_project()
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 111, in parse_project
>     self.evaluate_codeblock(self.ast, end=1)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 146, in evaluate_codeblock
>     raise e
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 140, in evaluate_codeblock
>     self.evaluate_statement(cur)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 151, in evaluate_statement
>     return self.function_call(cur)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 372, in function_call
>     return self.funcs[func_name](node, self.flatten(posargs), kwargs)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreterbase.py", line 55, in wrapped
>     return f(self, node, args, kwargs)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreter.py", line 1611, in func_project
>     self.add_languages(args[1:], True)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreter.py", line 1727, in add_languages
>     (comp, cross_comp) = self.detect_compilers(lang, need_cross_compiler)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\interpreter.py", line 1650, in detect_compilers
>     comp = self.environment.detect_c_compiler(False)
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\environment.py", line 376, in detect_c_compiler
>     p, out, err = Popen_safe([compiler, arg])
>   File "C:\Users\drako\Desktop\Projekte\vcpkg\downloads\tools\meson\meson-0.38.1\mesonbuild\mesonlib.py", line 497, in Popen_safe
>     o, e = p.communicate(write)
>   File "subprocess.py", line 801, in communicate
>   File "subprocess.py", line 1064, in _communicate
> IndexError: list index out of range

Problem is the same for x86 and x64.
This prevents me from installing gtk :-(

**Comment by codicodi** [comment]
Looks like there are some problems with Python, Meson and encodings. Following the traceback I suspect Meson calls opens pipe with `cl /?` to get compiler version, and Python incorrectly assumes output is in cp1252. What is your default codepage (you can run `chcp` in console if you're not sure) and default language used by compiler (the one that compiler uses when invoked on cmd line)? I'll try to reproduce this and report to Meson guys.

**Comment by Drako** [comment]
`chcp` says 850.
My system language is German (Germany) and compiler output is also German.

I also tested your workaround which worked out fine.

**Comment by codicodi** [comment]
Ok, I'll propose a patch bumping version of downloaded Meson when this gets fixed upstream

**Comment by JackBoosY** [comment]
Hi @Drako , thanks for reporting this issue!
I found nothing wrong with build libepoxy in latest version.
Could you update vcpkg and repro it?
Thx.

**Comment by JackBoosY** [comment]
Please open a new issue if this is still a problem for you.
Thx.
